app/Http/Controllers/SettingsController.php

```diff
@@ -18,9 +19,14 @@ public function profile() {
             $invoices = Auth::user()->invoices();
         }
 
+        $subscription = Auth::user()->subscriptions()->active()->first();
+        
+        $renews_at = $subscription ? (new Carbon($subscription->asStripeSubscription()->current_period_end))->toFormattedDateString() : False;
+
         return view('pages.profile', [
             'user' => Auth::user(),
-            'subscription' => Auth::user()->subscriptions()->active()->first(),
+            'subscription' => $subscription,
+            'subscription_renews_at' => $renews_at,
```

Comment:
I **think** `->current_period_end` should already be a Carbon object. If it is, I think you should be able to just do

```
$renews_at = $subscription->asStripeSubscription()->current_period_end->toFormattedDateString() ?? False;
```

Give it a try?